### PR TITLE
Update Google Analytics tags

### PR DESCRIPTION
**Pull request by sidane**
Google Analytics account details now come from ENV variables.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ class ApplicationController < ActionController::Base
   helper_method :current_lender
 
   def current_lender
-    current_user.lender
+    current_user.try(:lender)
```
Comment (nickcharlton) [inline app/controllers/application_controller.rb]: How come this is changing here?

File: app/helpers/google_analytics_helper.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+module GoogleAnalyticsHelper
+  def google_analytics(current_user:, current_lender:, domain:)
+    return unless ENV["GOOGLE_ANALYTICS_ID"]
+
+    user_type = current_user.try(:type)
```
Comment (nickcharlton) [inline app/helpers/google_analytics_helper.rb]: Is it common/likely that `current_user` won't exist such that we should use `.try`?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (sidane) [inline app/controllers/application_controller.rb]: So this comment sent me down a bit of a refactoring rabbit hole where I added `GuestUser` and `NoLender` null objects so we don't need `try` here (which is a code smell). The changes mean there is always a `current_user` and `current_lender` object, even if not logged in.

When not logged in, `current_user` is an instance of `GuestUser`. This user has no permissions, can't login and its lender is an instance of `NoLender`. `NoLender` has no name, loans or users.

This means we're not checking for nils anymore and simplifies the google analytics rendering quite a bit. It's a fair bit of refactoring just for rendering GA tags, but I think it's a good pattern to use moving forward.

What do you think @nickcharlton?

File: app/helpers/google_analytics_helper.rb (hunk 2 above)
Comment (sidane) [inline app/helpers/google_analytics_helper.rb]: When a visitor is not logged in then `current_user` is `nil`. So for example on the login page, there will be no `current_user` when rendering the google analytics tags.

However, the null object changes I've made (see other comment) mean this line is no longer present.

**Comment by sidane** [comment]
Replaced by #344 

We're going with a simpler google analytics implementation without any custom tracking of users, lenders and CSV/PDF download events.

**Pull request closed without merging**